backend/internal/api/router_test.go

```
@@ -0,0 +1,47 @@
+)
+
+func TestRouterCORSAllowedOrigins(t *testing.T) {
+	allowedOrigins := []string{
+		"http://localhost:3000",
+		"http://localhost:8082",
+	}
+	router := NewRouter(&Server{}, allowedOrigins)
+
+	for _, origin := range allowedOrigins {
+		t.Run(origin, func(t *testing.T) {
+			request := httptest.NewRequest(http.MethodOptions, "/health", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)